// vkz-grunt/dist/library/jshint.js
module.exports = {
    options: {
        curly: true,
        eqeqeq: true,
        undef: true,
        asi: true
    },
    gruntFiles: {
        options: {
            globals: {
                module: true,
                require: true,
                process: true
            },
            reporterOutput: ""
        },
        files: {
            src: ["Gruntfile.js"]
        }
    },
    source: {
        options: {
            browser: true,
            globals: {
                jQuery: true,
                angular: true,
                document: true,
                console: true,
                window: true,
                $: true,
                Camera: true,
                device: true,
                alert: true,
                cordova: true,
                google: true,
                "require": true,
                "process": true,
                "module" : true,
				Exception : true
            },
            reporterOutput: ""
        },
        files: {
            src: ["*.js", "src/**/*.js", "!**/bower_components/**"]
        }
    },
    test: {
        options: {
            browser: true,
            globals: {
                angular: true,
                "describe": true,
                "xdescribe": true,
                "before": true,
                "beforeEach": true,
                "after": true,
                "afterEach": true,
                "it": true,
                "xit": true,
                "inject": true,
                "expect": true,
                "spyOn": true,
                "jasmine": true,
                "module" : true
            },
            reporterOutput: ""
        },
        files: {
            src: ["test/**/*.js", "!**/bower_components/**"]
        }
    }
};
