// vkz-grunt/dist/web/config/concat.js

module.exports = {
	options: {
		separator: ';'
	},
	release: {
		files: [{
				src: ['src/bower_components/jquery/dist/jquery.js',
					'src/bower_components/jquery-ui/jquery-ui.min.js',
				],
				dest: 'www/js/jquery.js'
			},
			{
				src: ['src/bower_components/angular/angular.js'],
				dest: 'www/js/angular.js'
			},
			{
				src: [
					'src/bower_components/chart.js/dist/Chart.js',
					'src/bower_components/angular-chart.js/dist/angular-chart.js'
				],
				dest: 'www/js/chart.js'
			},
			{
				src: [
					'src/bower_components/bootstrap/dist/js/bootstrap.js',
				],
				dest: 'www/js/bootstrap.js'
			},
			{
				src: [
					'src/bower_components/**/*.js',
					'!src/bower_components/**/index.js',
					'!src/bower_components/**/Gruntfile.js',
					'!src/bower_components/**/*min.js',
					'!src/bower_components/bootstrap/**',
					'!src/bower_components/angular/**/*.js',
					'!src/bower_components/angular-i18n/**/*.js',
					'!src/bower_components/jquery/**/*.js',
					'!src/bower_components/jquery-ui/**/*.js',
					'!src/bower_components/chart.js/**/*.js',
					'!src/bower_components/angular-chart.js/**/*.js'
				],
				dest: 'www/js/libs.js'
			},
			{
				src: [
					'src/js/app.js',
					'src/**/*.js',
					'!src/bower_components/**/*.js'
				],
				dest: 'www/js/main.js'
			}
		]
	}
};
