// vkz-grunt/dist/web/config/cssmin.js
module.exports = {
    release: {
        files: {
            'www/css/main.css': [
                'src/css/*.css',
                'src/css/custom.css',
                'src/css/sass-custom.css',
                '!src/css/custom-*.css'
        ],
            'www/css/libs.css': [
                'src/bower_components/**/*.css',
                '!src/bower_components/angular/**/*.css',
                '!src/bower_components/bootstrap/**/*.css'
            ]
        },
        options: {
            aggressiveMerging: false,
            shorthandCompacting: false
        }
    },
    release_specific: {
        files: {
            'www/css/main.css': [
                'src/css/*.css',
                '!src/css/custom-*.css',
                '!src/css/custom.css',
                'src/css/custom-<%=grunt.task.current.args[0] %>.css'
            ],
            'www/css/libs.css': [
                'src/bower_components/**/*.css',
                '!src/bower_components/angular/**/*.css',
                '!src/bower_components/bootstrap/**/*.css'
            ]
        },
        options: {
            aggressiveMerging: false,
            shorthandCompacting: false
        }
    }
};
