// vkz-grunt/dist/web/config/htmlbuild.js

module.exports = {
	development: {
		src: 'src/index.html',
		dest: 'www/',
		options: {
			beautify: true,
			relative: true,
			styles: {
				main: 'www/css/**/*.css',
				libs: [
					'www/bower_components/**/*.css',
					'!www/bower_components/angular/**/*.css',
					'!www/bower_components/bootstrap/**/*.css'
				]
			},
			scripts: {
				jquery: ['www/bower_components/jquery/dist/jquery.js',
					'www/bower_components/jquery-ui/jquery-ui.js'
				],
				angular: ['www/bower_components/angular/angular.js'],
				chartjs: [
					'www/bower_components/chart.js/dist/Chart.js',
					'www/bower_components/angular-chart.js/dist/angular-chart.js'
				],
				libs: [
					'www/bower_components/**/*.js',
					'!www/bower_components/**/Gruntfile.js',
					'!www/bower_components/**/index.js',
					'!www/bower_components/**/package.js',
					'!www/bower_components/angular/**/*.js',
					'!www/bower_components/angular-i18n/**/*.js',
					'!www/bower_components/jquery/**/*.js',
					'!www/bower_components/jquery-ui/**/*.js',
					'!www/bower_components/bootstrap/grunt/*.js',
					'!www/bower_components/bootstrap/dist/js/npm.js',
					'!www/bower_components/chart.js/**/*.js',
					'!www/bower_components/angular-chart.js/**/*.js',
					'!www/bower_components/vkz-*/**/*.js'
				],
				development: ['www/debug_modules/stop-angular-overrides/stop-angular-overrides.js'],
				vkzlibs: ['www/bower_components/vkz-*/**/*.js'],
				main: [
					'www/js/app.js',
					'www/**/*.js',
					'!www/bower_components/**/*.js',
					'!www/debug_modules/**/*.js'
				]
			}
		}
	},
	release: {
		src: 'src/index.html',
		dest: 'www/',
		options: {
			beautify: true,
			relative: true,
			styles: {
				main: ['www/css/main.css', 'www/css/sass-custom.css'],
				libs: 'www/css/libs.css'
			},
			scripts: {
				jquery: 'www/js/jquery.js',
				angular: 'www/js/angular.js',
				chartjs: 'www/js/chart.js',
				libs: [
					'www/js/libs.js',
					'www/js/bootstrap.js'
				],
				development: [],
				vkzlibs: [],
				main: [
					'www/js/main.js',
					'www/js/partials.js'
				],
			}
		}
	},
};
