// vkz-grunt/dist/web/config/watch.js
module.exports = {
    programming: {
        files: [
            'src/**/*',
            '!src/bower_components/**/*',
            'src/bower_components/vkz-*/**/*'
        ],
        tasks: ['sync:development', 'postcss:development']
    },
    sassSupportedProgramming : {
        files: [
            'src/**/*',
            '!src/bower_components/**/*',
            'src/bower_components/vkz-*/**/*'
        ],
        tasks: ['sync:development', 'sass:debug', 'postcss:development']
    },
    programmingEfficient: {
        files: [
            'src/**/*',
            '!src/bower_components/**/*',
            'src/bower_components/vkz-*/**/*'
        ],
        tasks: ['sync:development', 'postcss:development']
    },
    sassSupportedProgrammingEfficient : {
        options: {
          // Options for the watch task go here
          interval: 2000
        },
        files: [
            'src/**/*',
            '!src/bower_components/**/*',
            'src/bower_components/vkz-*/**/*'
        ],
        tasks: ['sync:development', 'sass:debug', 'postcss:development']
    }
};
