// vkz-grunt/dist/app/config/concat.js
module.exports = {
    options: {
        separator: ';'
    },
    release: {
        files: [
            {
                src: ['src/versions/<%= grunt.task.current.args[0] %>/bower_components/jquery/dist/jquery.js'],
                dest: 'www/versions/<%= grunt.task.current.args[0] %>/js/jquery.js'
            },
            {
                src: ['src/versions/<%= grunt.task.current.args[0] %>/bower_components/angular/angular.js'],
                dest: 'www/versions/<%= grunt.task.current.args[0] %>/js/angular.js'
            },
            {
                src: [
                    'src/versions/<%= grunt.task.current.args[0] %>/bower_components/mobiscroll/js/*.js',
                    'src/versions/<%= grunt.task.current.args[0] %>/bower_components/mobiscroll/js/i18n/*.js',
                    '!src/versions/<%= grunt.task.current.args[0] %>/bower_components/mobiscroll/js/mobiscroll.jqmwidget.js'
                ],
                dest: 'www/versions/<%= grunt.task.current.args[0] %>/js/mobiscroll.js'
            },
            {
                src: [
                    'src/versions/<%= grunt.task.current.args[0] %>/bower_components/**/*.js',
                    '!src/versions/<%= grunt.task.current.args[0] %>/bower_components/angular-*/**/index.js',
                    '!src/versions/<%= grunt.task.current.args[0] %>/bower_components/angular/**/*.js',
                    '!src/versions/<%= grunt.task.current.args[0] %>/bower_components/angular-i18n/**/*.js',
                    '!src/versions/<%= grunt.task.current.args[0] %>/bower_components/mobiscroll/**/*.js',
                    '!src/versions/<%= grunt.task.current.args[0] %>/bower_components/jquery/**/*.js'
                ],
                dest: 'www/versions/<%= grunt.task.current.args[0] %>/js/libs.js'
            },
            {
                src: [
                    'src/versions/<%= grunt.task.current.args[0] %>/js/app.js',
                    'src/versions/<%= grunt.task.current.args[0] %>/**/*.js',
                    '!src/versions/<%= grunt.task.current.args[0] %>/bower_components/**/*.js',
					'!src/versions/<%= grunt.task.current.args[0] %>/lib/**/*.js' //Exclude all libraries because they often side load their own files on a onneed basis e.g. pdf.js and the rtfEditor.
                ],
                dest: 'www/versions/<%= grunt.task.current.args[0] %>/js/main.js'
            }
        ]
    },
    releaseShared: {
        files: [
            {
                src: ['src/shared/*.js', 'src/shared/**/*.js'],
                dest: 'www/shared/shared.js'
            }
        ]
    }
};
