// vkz-grunt/dist/app/config/copy.js
module.exports = function (grunt) {

    return {
        releaseVersion: {
            files: [
                {
                    expand: true,
                    cwd: 'src/versions/<%= grunt.task.current.args[0] %>/',
                    src: ['**/bower_components/angular-i18n/*.js'],
                    dest: 'www/versions/<%= grunt.task.current.args[0] %>/',
                    filter: function (filePath) {
                        return grunt.config('isSupportedLanguageFile')(filePath);
                    }
                },
                {
                    expand: true,
                    cwd: 'src/versions/<%= grunt.task.current.args[0] %>/',
                    src: ['verklizan.json', 'img/**'],
                    dest: 'www/versions/<%= grunt.task.current.args[0] %>/',
                },
				{
					expand: true,
					cwd: 'src/versions/<%= grunt.task.current.args[0] %>/lib',
					src: ['**'],
					dest: 'www/versions/<%= grunt.task.current.args[0] %>/js'
				}
            ]
        },
        releaseShared: {
            files: [
                {
                    expand: true,
                    cwd: 'src/shared',
                    src: ['**', '!**/*.js'],
                    dest: 'www/shared/'
                }
            ]
        },
        releaseIndex: {
            src: 'src/index.html',
            dest: 'www/index.html'
        },
        newRelease: {
            expand: true,
            cwd: 'src/versions/<%= grunt.config(\'config\').supportedServices[grunt.config(\'config\').supportedServices.length - 2] %>/', //length - 2 omdat array begint bij 0
            src: '**/*',
            dest: 'src/versions/<%= grunt.config(\'config\').targettedService %>/',
        },
        androidReleaseAPK: {
            expand:true,
            cwd: 'platforms/android/app/build/outputs/apk/release',
            src: ['app-release.apk'],
            dest: 'dist/android/<%= grunt.task.current.args[0] %>'
        }
    };
};
