// vkz-grunt/dist/app/config/htmlbuild.js
module.exports = function(grunt)  {
    var xml = require('node-xml-lite');
    var _INTAKER_APP_ = "Intaker";
    var _RESPONDER_APP_ = "Responer";

    var objectToExport = {
        targettedVersion: {
            src: 'src/versions/<%= grunt.task.current.args[0] %>/index.html',
            dest: 'www/versions/<%= grunt.task.current.args[0] %>/',
            options: {
                beautify: true,
                relative: true,
                styles: {
                    main: 'www/versions/<%= grunt.task.current.args[0] %>/css/**/*.css',
                    libs: 'www/versions/<%= grunt.task.current.args[0] %>/bower_components/**/*.css'
                },
                scripts: {
                    jquery: ['www/versions/<%= grunt.task.current.args[0] %>/bower_components/jquery/dist/jquery.js'],
                    angular: ['www/versions/<%= grunt.task.current.args[0] %>/bower_components/angular/angular.js'],
                    mobiscroll: [
                        'www/versions/<%= grunt.task.current.args[0] %>/bower_components/mobiscroll/js/*.js',
                        'www/versions/<%= grunt.task.current.args[0] %>/bower_components/mobiscroll/js/i18n/*.js',
                        '!www/versions/<%= grunt.task.current.args[0] %>/bower_components/mobiscroll/js/mobiscroll.jqmwidget.js'
                    ],
                    libs: [
                        'www/versions/<%= grunt.task.current.args[0] %>/bower_components/**/*.js',
                        '!www/versions/<%= grunt.task.current.args[0] %>/bower_components/angular-*/**/index.js',
                        '!www/versions/<%= grunt.task.current.args[0] %>/bower_components/angular/**/*.js',
                        '!www/versions/<%= grunt.task.current.args[0] %>/bower_components/angular-i18n/**/*.js',
                        '!www/versions/<%= grunt.task.current.args[0] %>/bower_components/mobiscroll/**/*.js',
                        '!www/versions/<%= grunt.task.current.args[0] %>/bower_components/jquery/**/*.js',
                        '!www/versions/<%= grunt.task.current.args[0] %>/bower_components/**/*.min.js'
                    ],
                    main: [
                        'www/versions/<%= grunt.task.current.args[0] %>/js/app.js',
                        'www/versions/<%= grunt.task.current.args[0] %>/**/*.js',
                        '!www/versions/<%= grunt.task.current.args[0] %>/bower_components/**/*.js'
                    ],
                    shared: ['www/shared/*.js', 'www/shared/**/*.js']
                }
            }
        },
        release: {
            src: 'src/versions/<%= grunt.task.current.args[0] %>/index.html',
            dest: 'www/versions/<%= grunt.task.current.args[0] %>/',
            options: {
                beautify: true,
                relative: true,
                styles: {
                    main: 'www/versions/<%= grunt.task.current.args[0] %>/css/main.css',
                    libs: 'www/versions/<%= grunt.task.current.args[0] %>/css/libs.css'
                },
                scripts: {
                    jquery: 'www/versions/<%= grunt.task.current.args[0] %>/js/jquery.js',
                    angular: 'www/versions/<%= grunt.task.current.args[0] %>/js/angular.js',
                    mobiscroll: 'www/versions/<%= grunt.task.current.args[0] %>/js/mobiscroll.js',
                    libs: 'www/versions/<%= grunt.task.current.args[0] %>/js/libs.js',
                    main: [
                        'www/versions/<%= grunt.task.current.args[0] %>/js/main.js',
                        'www/versions/<%= grunt.task.current.args[0] %>/js/partials.js'
                    ],
                    shared: ['www/shared/**/*.js']
                }
            }
        }
    };

    if(xml.parseString(grunt.file.read('config.xml')).childs[1].childs[0] === _INTAKER_APP_) {
    	objectToExport.targettedVersion.options.scripts.releaseLibs = [];
    	objectToExport.release.options.scripts.releaseLibs = ['www/versions/<%= grunt.task.current.args[0] %>/js/jSignature/jSignature.min.js',
    								                	'www/versions/<%= grunt.task.current.args[0] %>/js/mobiscroll/i18n/mobiscroll.i18n.nl.js',
    								                    'www/versions/<%= grunt.task.current.args[0] %>/js/PDFjs (viewer)/pdf.js',
    								                    'www/versions/<%= grunt.task.current.args[0] %>/js/wysiswygEditor/ckeditor.js',
    								                    'www/versions/<%= grunt.task.current.args[0] %>/js/wysiswygEditor/config.js'];
    }

    return objectToExport;
};
