// vkz-grunt/dist/app/config/shell.js
module.exports = function (grunt) {
    return {
        bowerInstallShared: {
            options: {
                stderr: false,
                execOptions:
                {
                    cwd: 'node_modules/.bin',
                }
            },
            command: function () {
                return 'bower install --config.cwd="' + grunt.config('paths').sharedAbsolute + '"';
            }
        },
        bowerInstallVersion: {
            options: {
                stderr: false,
                execOptions:
                {
                    cwd: 'node_modules/.bin',
                }
            },
            command: function (version) {
                var path = require('path');

                var versionPath = path.join(grunt.config('paths').versionsFolderAbsolute, version);

                return 'bower install --config.cwd="' + versionPath + '"';
            }
        },
        releaseAndroidCordova: {
            command: function () {
                return '"node_modules/.bin/cordova" build android --release --buildConfig config/build.json';
            }
        },
        releaseIOSCordova: {
            command: function () {
                return '"node_modules/.bin/cordova" build ios --release --device';
            },
            options: {
                execOptions: {
                    maxBuffer: Infinity
                }
            }
        },
        execute: {
            command: function () {
                return '<%= grunt.task.current.args[0] %>';
            }
        },
        installOnDeviceAndroid: {
            command: function (folder) {
                return 'adb install -r "dist/android/<%=grunt.task.current.args[0] %>/<%= grunt.config.data.appName %> v<%= grunt.config(\'config\').version %>.apk"';
            }
        },
        runAndroid: {
            command: function () {
                return '"node_modules/.bin/cordova" prepare android && "node_modules/.bin/cordova" run android';
            },
        },
        runiOS: {
            command: function () {
                return '"node_modules/.bin/cordova" run ios --device';
            },
            options: {
                execOptions: {
                    maxBuffer: Infinity
                }
            }
        },
        prepareHtmlbuild: {
            command: function() {
                return 'dos2unix src/versions/<%=grunt.task.current.args[0] %>/index.html';
            }
        }
    };
};
