// vkz-grunt/dist/app/config/sync.js
module.exports = function (grunt) {
    return {
        targettedVersion: {
            files: [
                {
                    expand: true,
                    cwd: 'src/versions/<%= grunt.task.current.args[0] %>/',
                    src: [
                        '**',
                        '!bower_components/**/*.json',
                        '!bower_components/**/*.map',
                        '!bower_components/**/*.gzip',
                        '!bower_components/**/*.md',
                        '!bower_components/**/*.min.js',
                        '!bower_components/**/*.txt',
                        '!bower_components/**/*.html',
                        '!bower_components/**/src/**',
                        'bower_components/vkz-*/**/*'
                    ],
                    dest: 'www/versions/<%= grunt.task.current.args[0] %>/',
                }
            ],
            ignoreInDest: '**/config.json',
            updateAndDelete: true
        },
        syncShared: {
            files: [
                {
                    expand: true,
                    cwd: 'src/shared/',
                    src: [
                        '**',
                        '!bower_components/**/*.json',
                        '!bower_components/**/*.map',
                        '!bower_components/**/*.gzip',
                        '!bower_components/**/*.md',
                        '!bower_components/**/*.min.js',
                        '!bower_components/**/*.txt',
                        '!bower_components/**/*.html',
                        '!bower_components/**/src/**'
                    ],
                    dest: 'www/shared/',
                }
            ],
            ignoreInDest: '**/config.json',
            updateAndDelete: true
        },
        syncRoot: {
            files: [
                {
                    expand: true,
                    cwd: 'src/',
                    src: [
                        '*.*',
                    ],
                    dest: 'www',
                }
            ],
            ignoreInDest: '**/config.json',
            updateAndDelete: false
        }
    };
};
