// vkz-grunt/dist/config examples/library/karma.conf.js
module.exports = function (config) {
    config.set(
        {
            basePath: '.',
            frameworks: [
                'jasmine'
            ],
            files: [
                'bower_components/jquery/dist/jquery.js',
                'bower_components/angular/angular.js',
                'bower_components/jasmine-promise-matchers/dist/jasmine-promise-matchers.js',
                'node_modules/angular-mocks/angular-mocks.js',
                'src/**/*.js',
                'test/matchers/**/*.js',
                'test/unit/**/*.js'
            ],
            singleRun: true,
            browsers: ['Chrome'],
            reporters: ['progress', 'coverage'],
            preprocessors: {
                'src/**/*.js': ['coverage'] //'src//**/*.js'
            },
            coverageReporter: {
                dir: 'js.coverage/',
                reporters: [
                    { type: 'html', subdir: 'report-html' },
                    { type: 'text-summary' },
                    { type: 'json', subdir: 'report-json' },
                ]
            },
            plugins: [
                'karma-ng-scenario',
                'karma-chrome-launcher',
                'karma-jasmine',
                'karma-jasmine-html-reporter',
                'karma-coverage',
            ],
            autoWatch: true
        });
    };
